scripts/generate_dump

```diff
@@ -758,10 +758,20 @@ enable_logrotate() {
 #  None
 ###############################################################################
 collect_mellanox() {
-    local sai_dump_filename="/tmp/sai_sdk_dump_$(date +"%m_%d_%Y_%I_%M_%p")"
+    local sai_dump_folder="/tmp/saisdkdump"
+    local sai_dump_filename="${sai_dump_folder}/sai_sdk_dump_$(date +"%m_%d_%Y_%I_%M_%p")"
+
+    ${CMD_PREFIX}docker exec -it syncd mkdir -p $sai_dump_folder
     ${CMD_PREFIX}docker exec -it syncd saisdkdump -f $sai_dump_filename
-    ${CMD_PREFIX}docker exec syncd tar Ccf $(dirname $sai_dump_filename) - $(basename $sai_dump_filename) | tar Cxf /tmp/ -
-    save_file $sai_dump_filename sai_sdk_dump true
+
+    copy_from_docker syncd $sai_dump_folder $sai_dump_folder
+    echo "$sai_dump_folder"
+    for file in `ls $sai_dump_folder`; do
+        save_file ${sai_dump_folder}/${file} sai_sdk_dump true 
+    done
+
+    ${CMD_PREFIX}rm -rf $sai_dump_folder
+    ${CMD_PREFIX}docker exec -it syncd rm -rf $sai_dump_folder
 
     local mst_dump_filename="/tmp/mstdump"
```

Comment:
@DavidZagury can you please check if the mstdump is also needed here or already covered by the SAI dump once the extended API is called.

Reply:
@liat-grozovik,

As you see the mst dump is not part of the extended SAI dump.
the change that we can do is to create the mst logs into the same dir for the SAI dump but it will only save one code line "save_file "${mst_dump_filename}${i}" mstdump true" with the same performance :)
I don't think its a necessary change.

Reply:
Since the mst files are not just gathered here to the tar file but are also being created in this script by SONiC (and not by SAI), it would mean to move the mst code to before the new loop and changing it to this:
```
local max_dump_count="3"
for i in $(seq 1 $max_dump_count); do
        ${CMD_PREFIX}/usr/bin/mstdump /dev/mst/mt*conf0 &gt; "${sai_dump_folder}${i}"
done
```
This would also change the location that the mst files are being saved into.
It wouldn't be a significant change, and would not really make the code any simpler.

Reply:
If SAI will add this to the file they are creating we could remove all this code.

Reply:
Agree it can be later enhanced. For now LGTM